sdk/src/main/scala/coop/rchain/sdk/error/FatalError.scala

```diff
@@ -0,0 +1,35 @@
+package coop.rchain.sdk.error
+
+import coop.rchain.sdk.syntax.all._
+
+/**
+  * Represents the error which should not be handled and should cause the main process to exit.
+  *
+  * It extends [[Throwable]] directly to indicate that is "lower" level error then [[Exception]].
+  *
+  * Note: it's still caught by [[scala.util.control.NonFatal]] extractor.
+  */
+final class FatalError(message: String) extends Throwable(message) {
```

Comment:
Is this a more explicit alternative to `assert`?

Reply:
Using `assert` is in some way cheating because we don't allow Java `throw` errors, but with assert this is possible.

This error is just more specific error to signal what should not be caught by the caller and basically requires exit of the process.